src/hashtable.py

```diff
@@ -51,8 +51,45 @@ def insert(self, key, value):
 
         Fill this in.
         '''
-        pass
-
+        # Create hash index to fit in the table
+        idx = self._hash_mod(key)
+        # Identify the target slot for storage
+        target_slot = self.storage[idx]
+        # Identify and store the item hashed at that slot if any
+        # this gives us a mutable variable to iterate over later when going
+        # down the linked list
+        node = target_slot
+        # Create a new list node out of the key and value so that we can chain as necessary
+        new_node = LinkedPair(key, value)
+        # If our target slot already has a node in it...
+        if target_slot is not None:
+            # and if that node has the same key as our new node...
+            if target_slot.key == new_node.key:
+                # override that node with the new node
+                new_node.next = target_slot.next
+                self.storage[idx] = new_node
+            # otherwise if they do not share a key...
+            else:
+                # iterate down the linked list
+                while node is not None:
+                    # check to see if there is another node after the current one
+                    if node.next is None:
+                        # if not link the new node onto the tail of the list
+                        node.next = new_node
+                        break
+                    # If there is another node linked, and that node shares a key
+                    elif node.next.key == new_node.key:
```

Comment:
same override/update for the head/first node check, but for the second node/head.next/ 2nd pointer!